### PR TITLE
Travis CI Fix

### LINKED ISSUE #87: Update Travis CI to track Dependabot updates
*This issue is part of several tasks listed in #79.*

### PR DESCRIPTION
**Pull request by Quantalabs**
# Travis CI Fix
___
## Known Issues
In #86 and #83, changes to requirements were made for integrating Dependabot into epispot, causing Travis CI to install wrong versions. This PR fixes that, and makes sure Travis CI installs the correct version based on the updates Dependabot gives.
## Code Breakdown
`requirements/`:
  - This moves all requirements into the folder, however, the bin requirements are now deleted.
 
`scripts/`:
  - These are the build scripts for different environments, including conda, and Travis CI.
  - The `travis` folder includes scripts for Travis CI, which will be extended in future PRs.

`travis.yml`:
   - Use the Travis build script to install dependencies, and then run tests.

## Additional Notes
Closes #87

### CODE REVIEW

File: .travis.yml
Comment (quantum9Innovation) [inline .travis.yml:4]: Why are we deprecating Python 3.6?

File: scripts/README.md
Comment (quantum9Innovation) [inline scripts/README.md:1]:
```suggestion
# scripts/
```
Don't wrap filenames in code

File: scripts/README.md
Comment (quantum9Innovation) [inline scripts/README.md:2]:
```suggestion
The scripts/ folder contains an array of scripts used for building and uploading packages.
```

File: scripts/README.md
Comment (quantum9Innovation) [inline scripts/README.md:8]:
```suggestion
| Script | Description |
| --- | --- |
| ![build.sh](./build.sh) | Installs requirements and sets up project. |
| ![conda.sh](./conda.sh) | Equivalent of build.sh for conda environments. |
| ![travis/](./travis) | Travis CI config |
```

File: requirements/README.md
Comment (quantum9Innovation) [inline requirements/README.md:1]:
```suggestion
# requirements/
```

File: requirements/README.md
Comment (quantum9Innovation) [inline requirements/README.md:8]:
```suggestion
| File | Description |
| --- | --- |
| requirements.txt | Base requirements for the package. |
| requirements-nightly.txt | Base requirements plus experimental dependencies for the nightly package. |
| requirements-dev.txt | All the base requirements and experimental dependencies plus dev tools. |
```

File: scripts/build.sh
Comment (quantum9Innovation) [inline scripts/build.sh:1]:
```suggestion
echo "Installing requirements... hold tight."
```

File: scripts/build.sh
Comment (quantum9Innovation) [inline scripts/build.sh:3]:
```suggestion
echo "Installation complete."

```

File: scripts/conda.sh
Comment (quantum9Innovation) [inline scripts/conda.sh:1]:
```suggestion
echo "Creating conda env `epispot`..."
```
Make sure to tell the user what environment is being created.

File: scripts/conda.sh
Comment (quantum9Innovation) [inline scripts/conda.sh:2]:
```suggestion
conda create -n epispot
```
Use the user-selected default Python version. After all, epispot supports Python 3.6-3.9, so we don't need to worry about this.

File: scripts/conda.sh
Comment (quantum9Innovation) [inline scripts/conda.sh:6]:
```suggestion
echo "Installing requirements with `pip`..."
```
Tell the user that the requirements are being installed via `pip` instead of `conda` to facilitate dependency management.

File: scripts/conda.sh
Comment (quantum9Innovation) [inline scripts/conda.sh:9]:
```suggestion
echo "Installation complete."
```
Keep success messages consistent across build scripts.

File: scripts/travis/build.sh
Comment (quantum9Innovation) [inline scripts/travis/build.sh:3]:
```suggestion
```
If it has a version, don't install it directly. Such dependencies won't be updated by Dependabot and thus tests won't be accurate. If this is due to a dependency conflict with `SciencePlots`, install requirements/requirements.txt and *then* install the dev requirements.

File: scripts/travis/build.sh
Comment (quantum9Innovation) [inline scripts/travis/build.sh:5]:
```suggestion
echo "Finished installing dependencies."

```

**Review by quantum9Innovation — CHANGES_REQUESTED** [review]
Please review all comments thoroughly before we consider a merge strategy.

File: .travis.yml
Comment (Quantalabs) [reply .travis.yml:4]: According to the dependency requirements, the matplotlib version no longer supports 3.6.

**Review by quantum9Innovation — APPROVED** [review]
Do not merge yet; we need to wait until all checks have passed.

**Comment by quantum9Innovation** [comment]
Merge with these exact criteria:

Use the **Squash and merge** strategy.
Change the commit title to **"Fix Travis CI (#88)"**—no quotes, no bold styling
Use the default commit message—**do not change**

**Pull request merged**
